### PR TITLE
Convert migration script from bash to javascript

### LINKED ISSUE #45: Add group creation to server migration script
The server migration script should create all Sinopia groups w/ default control/read ACLs so that they're there later on.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #45

Populates an empty trellis with root, repository, and ld4p group containers, plus the root ACLs we need

### CODE REVIEW

File: config/default.js [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+  // webid list for users that will get acl:Control permissions for all groups
+  adminUsers: defer(function() {
+    return [
+      '62748a6a-3a53-4b4c-976d-aac6062d594b',
+      '90c16e02-3fdd-4459-968e-f4795b682bd0',
+      '11a584b8-2ae5-4fd1-b8a1-e28976c48132',
+      '56b84999-793f-4fa6-ad20-be8f62a5c2de',
+      '2586583c-44bd-4d3c-9a6a-0d94510be5ea'
+    ].map(iss => {
+      return `https://cognito-idp.us-west-2.amazonaws.com/${this.userPoolId}/${iss}`
```
Comment (ndushay) [inline config/default.js]: you're mapping the uuids for the users;  the iss is more like the base url.  this relates to work I'm doing right now.

Also, ima gonna have a config variable for webidBaseUrl

Comment (ndushay) [reply config/default.js]: but this all relates to ... what is the plan for the ACL work ... today?

Comment (mjgiarlo) [reply config/default.js]: ah! I confused `iss` and `sub`. Will change.

Comment (mjgiarlo) [reply config/default.js]: done.

File: config/default.js [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+
 module.exports = {
-  adminUsers: // webid list for users that will get acl:Control permissions for all groups
-    [
-      'http://sinopia.io/users/webid4Admin1',
-      'http://sinopia.io/users/webid4Admin2'
-    ],
+  // webid list for users that will get acl:Control permissions for all groups
+  adminUsers: defer(function() {
+    return [
```
Comment (mjgiarlo) [inline config/default.js]: This is a list of Cognito `sub` values identifying the devs and Michelle within the dev Sinopia environment.

File: config/test.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
 module.exports = {
-  testUser: 'sinopia-devs_client-tester'
+  testUser: 'sinopia-devs_client-tester',
+  // override the default value for testing
+  adminUsers: [
```
Comment (mjgiarlo) [inline config/test.js]: Override the above value so the tests pass. The below values are hard-coded in our test fixtures.

File: fixtureWAC/basicAuth/migration.sh [hunk 4]
```diff
@@ -1,46 +0,0 @@
-#!/usr/bin/env bash
```
Comment (mjgiarlo) [inline fixtureWAC/basicAuth/migration.sh]: This has been replaced with `src/populateEmptyTrellis.js`.

File: fixtureWAC/rootWAC.mustache [hunk 5]
```diff
@@ -0,0 +1,20 @@
+@prefix acl: <http://www.w3.org/ns/auth/acl#> .
+@prefix foaf: <http://xmlns.com/foaf/0.1/> .
+
+<http://platform:8080/#control> a acl:Authorization ;
+        acl:mode      acl:Read ;
+        acl:mode      acl:Write ;
+        acl:mode      acl:Control ;
+{{{adminAgents}}}
```
Comment (mjgiarlo) [inline fixtureWAC/rootWAC.mustache]: Using a simple mustache template to inject webids from config.

File: fixtureWAC/rootWAC.mustache [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        acl:accessTo  <http://platform:8080/> .
+
+<http://platform:8080/#write> a acl:Authorization ;
+        acl:mode       acl:Read ;
+        acl:mode       acl:Write ;
+        acl:agentClass acl:AuthenticatedAgent ;
```
Comment (mjgiarlo) [inline fixtureWAC/rootWAC.mustache]: Grant write to all users in our Cognito pool with valid JWTs.

File: jest.config.js [hunk 7]
```diff
@@ -30,7 +30,9 @@ module.exports = {
 
   // An array of glob pattern strings used to skip coverage collection
   coveragePathIgnorePatterns: [
-    'src/cli/commander.js'
+    'src/cli/commander.js',
+    'src/cli/authenticate.js',
```
Comment (mjgiarlo) [inline jest.config.js]: Ignore authN CLI and migration script from code coverage calculations.

File: package.json [hunk 8]
```diff
@@ -11,6 +11,7 @@
     "jest-cov": "jest --colors --coverage --runInBand",
     "jest-ci": "jest --colors --ci --coverage --runInBand",
     "jest-integration": "jest --colors --testRegex=\\.integration\\.js$",
+    "migration": "babel-node src/populateEmptyTrellis.js",
```
Comment (mjgiarlo) [inline package.json]: Make the migration an npm script.

File: src/populateEmptyTrellis.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+request('POST', `${config.baseUrl}/repository`, {
+  headers: {
+    'Content-Type': 'text/turtle',
+    'Link': '<http://www.w3.org/ns/ldp#BasicContainer>; rel="type"',
+    'Slug': 'ld4p'
+  },
+  body: fs.readFileSync('./fixtureWAC/groupLd4pContainer.ttl')
+})
+
+console.log('setting root container ACLs')
```
Comment (mjgiarlo) [inline src/populateEmptyTrellis.js]: Do ACLs after creating containers, else we need a valid JWT for the above operations.

File: config/default.js (hunk 2 above)
Comment (ndushay) [reply config/default.js]: you might indicate if these are real sub values, and for which users -- this is a bit ... opaque.

File: config/default.js [hunk 10]
```diff
@@ -1,9 +1,19 @@
...
+      '62748a6a-3a53-4b4c-976d-aac6062d594b',
+      '90c16e02-3fdd-4459-968e-f4795b682bd0',
+      '11a584b8-2ae5-4fd1-b8a1-e28976c48132',
+      '56b84999-793f-4fa6-ad20-be8f62a5c2de',
+      '2586583c-44bd-4d3c-9a6a-0d94510be5ea'
+    ].map(sub => {
+      return `https://cognito-idp.us-west-2.amazonaws.com/${this.userPoolId}/${sub}`
```
Comment (ndushay) [inline config/default.js]: consider making the baseurl its own config value???

File: src/populateEmptyTrellis.js (hunk 9 above)
Comment (ndushay) [reply src/populateEmptyTrellis.js]: I would add your PR comment as a code comment - it's helpful.

File: config/default.js (hunk 2 above)
Comment (mjgiarlo) [reply config/default.js]: will do!

File: config/default.js (hunk 10 above)
Comment (mjgiarlo) [reply config/default.js]: on its way

File: src/populateEmptyTrellis.js (hunk 9 above)
Comment (mjgiarlo) [reply src/populateEmptyTrellis.js]: on it

**Comment by mjgiarlo** [comment]
@ndushay all changes made. thanks!

**Comment by mjgiarlo** [comment]
thx, @ndushay!

**Pull request merged**